exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/References.kt

```diff
@@ -236,41 +237,88 @@ private fun <ID : Comparable<ID>> List<Entity<ID>>.preloadRelations(
         }
     }
 
+    fun Entity<*>.getReferenceId(
+        delegateRefColumn: Column<*>,
+        refColumns: Map<Column<*>, Column<*>>,
+        isNotCompositeIdReference: Boolean
+    ): Any? {
+        return if (isNotCompositeIdReference) {
+            delegateRefColumn.lookup()
+        } else {
+            val childValues = refColumns.keys.map { it.lookup() }
+            if (childValues.any { it == null }) {
+                null
+            } else {
+                CompositeID {
+                    refColumns.values.forEachIndexed { i, parent ->
```

Comment:
I feel like it would be good idea to add a function to build `CompositeID`. This piece of code I was able to extract to 

```kotlin
private fun compositeId(entries: List<Pair<Column<EntityID<Comparable<Any>>>, Comparable<Any>>>) = CompositeID {
    entries.forEachIndexed { i, entry ->
        it[entry.first] = entry.second
    }
}
```

so the call of it looks like 

```kotlin
compositeId(refColumns.values.zip(childValues) as List<Pair<Column<EntityID<Comparable<Any>>>, Comparable<Any>>>)
```

Typing unfortunately looks already not a human readable a bit =)